engine/runtime/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -522,11 +524,8 @@ class Compiler(
             .map { concreteBindings =>
               concreteBindings
             }
-          val ir = context.getIr(module)
-          val currentLocal = ir.unsafeGetMetadata(
-            BindingAnalysis,
-            "Synthetic parsed module missing bindings"
-          )
+          ensureParsed(module)
```

Comment:
Another `ensureParsed` for a synthetic module? If hte serialization of synthetic modules would not be skipped, could this be removed? See my other comment.

Reply:
Probably not - [according to Hubert](https://github.com/enso-org/enso/pull/8160/files#r1374425774).

Reply:
Probably it wasn't that great idea to remove it. We are getting [failure in Numpy project test](https://github.com/enso-org/enso/actions/runs/6667090478/job/18119889353?pr=8160#step:10:4503) on Linux and Mac.

Reply:
After [bcad8e4](https://github.com/enso-org/enso/pull/8160/commits/bcad8e4dba7eb46a147e09730f784bfc7e557bdb) the _CI is  green_ again. I guess that explains why `ensureParsed` is needed here (at least for current moment).